Update General: JSON (#789)
Codecademy/docs: content/general/concepts/json/json.md

  - 'paths/full-stack-engineer-career-path'
---

<link rel="canonical" href="https://www.codecademy.com/article/what-is-json" />

**JavaScript Object Notation (JSON)** is a data format that is readable, writable, and parsable for both humans and machines. JSON is based on the syntax of the third edition of [JavaScript](https://www.codecademy.com/resources/docs/javascript) ([ECMA-262](https://www.ecma-international.org/publications-and-standards/standards/ecma-262/)) that was released back in December 1999. However, it is language-independent.

> JSON is commonly used for transferring data between clients and servers for tasks such as web browsing or form submission. Companies also use JSON to allow their data to used in other applications via [API](https://www.codecademy.com/resources/docs/general/api). Some example include:

Reviewer: ```suggestion
JSON is commonly used for transferring data between clients and servers for tasks such as web browsing or form submission. Companies also use JSON to allow their data to use in other applications via [API](https://www.codecademy.com/resources/docs/general/api). Some examples include:
```

> - [google maps]: https://developers.google.com/maps/documentation/geocoding/start

Reviewer: ```suggestion
- [google maps](https://developers.google.com/maps/documentation/geocoding/start)
```

}
```

> Data is stored in name-value pairs in object- separated by a colon `:`. The pairs themselves are separated by a comma `,`. The following [data types](https://www.codecademy.com/resources/docs/general/data-types) can be used:

Reviewer: ```suggestion
Data is stored in an object, depicted by two curly braces `{``}`, and name-value pairs are separated by a colon `:`. The pairs themselves are separated by a comma `,`. The following are [data types](https://www.codecademy.com/resources/docs/general/data-types) that can be used:
```

]
```

A few important points to note about JSON syntax:

- While it is derived from from the JavaScript language, JSON itself is not JavaScript.

> - Trailing commas are forbidden

Reviewer: ```suggestion
- Trailing commas are forbidden.
```

> - Even thought JavaScript names are not this strict, JSON property names _must_ be in double-quotes.

Reviewer: ```suggestion
- Although JavaScript names are not this strict, JSON property names must be in double quotes.
```

## Bad Practices

Below are a few points regarding what should be avoided when using JSON format:

> - It is not useful to abide by an XML-like, "attribute vs. element" framework. JSON is only about "name-value" pairs.

Reviewer: ```suggestion
- It is not useful to abide by an XML-like, "attribute vs. element" framework. JSON is only about name-value pairs.
```

- While the nesting of inner-JSON objects is possible, going more than two levels deep could get complicated.

## Example

The following JSON example uses one level of nested objects, an array, and each valid data type:

```pseudo
{
  "student": {
    "name": "Codey",
    "age": 30,
    "fullTime": true,

> 

Reviewer: ```suggestion
```
